### PR TITLE
Support Paraformer-zh in optimum-intel

### PR DESCRIPTION
**Pull request by padatta**
## Summary

This PR adds support for exporting Alibaba's Paraformer ASR model (`funasr/paraformer-zh`) to OpenVINO IR format with comprehensive INT8 quantization capabilities **and full inference support**.

## Key Features

- **Full Paraformer model export** to OpenVINO IR format (FP16)
- **INT8 full quantization** with AISHELL-1 dataset calibration
- **GPU-compatible** indexed operations using `torch.where` and `index_fill_`
- **Per-tensor activation quantization** for optimal inference performance
- **Auto-detection** of Paraformer models via characteristic files (`am.mvn`, `config.yaml`, `tokens.json`)
- **OpenVINO inference model** (`OVParaformerForSpeechSeq2Seq`) following the SpeechT5TTS pattern

### Export Implementation
| File | Change |
|------|--------|
| `optimum/exporters/openvino/modeling_paraformer.py` | **NEW** - Complete export implementation (2118 lines) |
| `optimum/exporters/openvino/__main__.py` | INT8 quantization pipeline with NNCF integration |
| `optimum/commands/export/openvino.py` | CLI command updates for paraformer library support |
| `optimum/intel/openvino/utils.py` | Added `aishell-1` to predefined speech-to-text datasets |
| `optimum/intel/utils/modeling_utils.py` | Paraformer model auto-detection |

### Inference Implementation  
| File | Change |
|------|--------|
| `optimum/intel/openvino/modeling_speech2text.py` | **NEW** - Full inference implementation (847 lines) |
| `optimum/intel/__init__.py` | Export OVParaformerForSpeechSeq2Seq |
| `optimum/intel/openvino/__init__.py` | Import from modeling_speech2text |
| `optimum/intel/pipelines/accelerator_utils.py` | Add to ASR pipeline support |
| `optimum/intel/utils/dummy_openvino_objects.py` | Add dummy object |

### Export Models
```bash
# FP16 Export
optimum-cli export openvino --trust-remote-code --model funasr/paraformer-zh ov_paraformer_fp16

# INT8 Full Quantization
optimum-cli export openvino --trust-remote-code --model funasr/paraformer-zh \
    --quant-mode int8 --dataset aishell-1 ov_paraformer_int8
```
## Inference with OVParaformerForSpeechSeq2Seq
from optimum.intel.openvino import OVParaformerForSpeechSeq2Seq
import torch

### Load model
model = OVParaformerForSpeechSeq2Seq.from_pretrained(
    "ov_paraformer_int8/ov_models",
    device="GPU"
)

### Prepare inputs
speech = torch.randn(1, 100, 560)  # [batch, time, features]
speech_lengths = torch.tensor([100], dtype=torch.int32)

### Run inference
output = model(speech, speech_lengths)

### Get results
token_ids = output.token_ids  # Decoded token IDs
token_num = output.token_num  # Number of valid tokens
logits = output.logits        # Raw logits

### CODE REVIEW

**Review by openvino-agent — CHANGES_REQUESTED** [review]
t

**Review by rkazants — CHANGES_REQUESTED** [review]
Please add tests,
openvino.py, main.py should not be changed. Please check what files are changed when adding new models support in other PRs.
Thanks

File: optimum/exporters/openvino/modeling_paraformer.py [hunk 1]
```diff
@@ -0,0 +1,2315 @@
...
+
+        encoder_outs = self.model.encoders(xs_pad, mask)
+        xs_pad, masks = encoder_outs[0], encoder_outs[1]
+
+        xs_pad = self.model.after_norm(xs_pad)
+
+        if self.ctc_linear is not None:
+            xs_pad = self.ctc_linear(xs_pad)
+            xs_pad = F.softmax(xs_pad, dim=2)
+
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/exporters/openvino/modeling_paraformer.py:1273]: Fixed. Added [import torch.nn.functional as F] at the top of the file.

File: tests/openvino/utils_tests.py [hunk 2]
```diff
@@ -213,6 +213,7 @@
     "wav2vec2-hf": "optimum-intel-internal-testing/tiny-random-Wav2Vec2Model",
     "wav2vec2-conformer": "optimum-intel-internal-testing/tiny-random-wav2vec2-conformer",
     "whisper": "optimum-intel-internal-testing/tiny-random-whisper",
+    "paraformer": "funasr/paraformer-zh",
```
[the remark this answers is not in the document]
Comment (padatta) [reply tests/openvino/utils_tests.py:216]: Fixed. Gated behind RUN_SLOW_EXPORT_TESTS=1 environment variable. The paraformer model entry is only included when explicitly enabled.

File: tests/openvino/test_exporters_cli.py [hunk 3]
```diff
@@ -113,6 +114,7 @@ class OVCLIExportTestCase(unittest.TestCase):
         ("text-to-video", "ltx-video"),
         ("feature-extraction", "sam"),
         ("text-to-audio", "speecht5"),
+        ("automatic-speech-recognition", "paraformer"),
         ("zero-shot-image-classification", "clip"),
     ]
```
[the remark this answers is not in the document]
Comment (padatta) [reply tests/openvino/test_exporters_cli.py:119]: Fixed. Moved to conditional append gated behind RUN_SLOW_EXPORT_TESTS=1.

File: tests/openvino/test_export.py [hunk 4]
```diff
@@ -79,6 +80,7 @@ class ExportModelTest(unittest.TestCase):
         "roberta": OVModelForTokenClassification,
         "wav2vec2": OVModelForAudioClassification,
         "whisper": OVModelForSpeechSeq2Seq,
+        "paraformer": OVParaformerForSpeechSeq2Seq,
```
[the remark this answers is not in the document]
Comment (padatta) [reply tests/openvino/test_export.py:83]: Fixed. Conditionally added only when RUN_SLOW_EXPORT_TESTS=1 is set.

File: optimum/exporters/openvino/paraformer_plugin.py [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+            np.random.seed(42)
+            for i in range(num_samples):
+                noise = np.random.randn(*base_features.shape).astype(np.float32) * (0.01 + i * 0.0004)
+                features = base_features + noise
+                
+                speech = features[np.newaxis, :].astype(np.float32)
+                speech_lengths = np.array([features.shape[0]], dtype=np.int32)
+                # Use 'speech.1' as the model input name (from OV conversion)
+                calibration_samples.append({'speech.1': speech, 'speech_lengths': speech_lengths})
+        else:
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/exporters/openvino/paraformer_plugin.py:382]: Fixed. Now dynamically resolves input names from [ovm.inputs] using [get_any_name()], matching speech and length inputs by keyword pattern.

File: optimum/exporters/openvino/paraformer_plugin.py [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
+        logger.debug("Patched main_export to support Paraformer models")
+        
+    except Exception as e:
+        logger.warning(f"Could not patch main_export for Paraformer support: {e}")
+
+
+# Auto-register when this module is imported
+register_paraformer_with_tasks_manager()
+patch_main_export()
+patch_main_quantize()
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/exporters/openvino/paraformer_plugin.py:609]: Fixed. Replaced auto-registration at import time with an explicit [initialize_paraformer_support()]function, called from [model_configs.py] during initialization.

File: optimum/intel/pipelines/accelerator_utils.py [hunk 7]
```diff
@@ -128,6 +129,12 @@ def get_openvino_model_class(
             config = AutoConfig.from_pretrained(model_id, **hub_kwargs)
         if any(arch.endswith("ForCTC") for arch in config.architectures):
             ov_model_class = OV_TASKS_MAPPING[task][0]
+        # Check for Paraformer models - detected by model_type or architecture
+        elif (
+            getattr(config, "model_type", "").lower() == "paraformer"
+            or any("Paraformer" in arch for arch in getattr(config, "architectures", []))
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/intel/pipelines/accelerator_utils.py:135]: Fixed. Now uses [architectures = getattr(config, "architectures", []) or []] to safely handle None values.

File: optimum/intel/openvino/modeling_speech2text.py [hunk 8]
```diff
@@ -0,0 +1,850 @@
...
+            # Fall back to first input
+            speech_input_name = list(self.input_names.keys())[0]
+        
+        # Prepare inputs
+        speech_np = speech.cpu().numpy() if isinstance(speech, torch.Tensor) else speech
+        lengths_np = speech_lengths.cpu().numpy() if isinstance(speech_lengths, torch.Tensor) else speech_lengths
+        
+        inputs = {
+            speech_input_name: speech_np,
+            "speech_lengths": lengths_np,
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/intel/openvino/modeling_speech2text.py:660]: Fixed. Added dynamic resolution of the lengths input name by searching [self.input_names] for entries containing "length", with fallback to non-speech inputs, then to "speech_lengths".

File: optimum/intel/openvino/modeling_speech2text.py [hunk 9]
```diff
@@ -0,0 +1,850 @@
...
+        Args:
+            save_directory: Directory to save model files
+        """
+        save_path = Path(save_directory)
+        save_path.mkdir(parents=True, exist_ok=True)
+        
+        if self._single_model:
+            model_path = save_path / OV_XML_FILE_NAME
+            openvino.save_model(self.model, str(model_path))
+            logger.info(f"Saved model to {model_path}")
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/intel/openvino/modeling_speech2text.py:783]: Fixed. Added a guard that raises [ValueError] with a clear message when [save_pretrained()] is called in [compile_only] mode.

File: optimum/exporters/openvino/modeling_paraformer.py [hunk 10]
```diff
@@ -0,0 +1,2315 @@
...
+                add_file_root_path(model_or_path, conf_json["file_path_metas"], cfg)
+            cfg.update(kwargs)
+            if "config" in cfg:
+                config = OmegaConf.load(cfg["config"])
+                kwargs = OmegaConf.merge(config, cfg)
+                kwargs["model"] = config["model"]
+    elif os.path.exists(os.path.join(model_or_path, "config.yaml")):
+        config = OmegaConf.load(os.path.join(model_or_path, "config.yaml"))
+        kwargs = OmegaConf.merge(config, kwargs)
+        init_param = os.path.join(model_or_path, "model.pt")
```
[the remark this answers is not in the document]
Comment (padatta) [reply optimum/exporters/openvino/modeling_paraformer.py:1974]: Fixed. Made [omegaconf] an optional import with try/except guard and [_OMEGACONF_AVAILABLE] flag. [download_from_hf()] now raises a clear [ImportError] when [omegaconf] is not installed.

**Review by rkazants — CHANGES_REQUESTED** [review]
incorrectly added support to optimum-intel:
- double check how others TTS models are added (Qwen3-TTS, Kokoro). Architecture and the whole structure of optimum-intel should not be broken
- avoid new test suites added like test_paraformer and others. We should use existing test suites
- no int4 int8 compression results, GPU results?

**Comment by padatta** [comment]
> incorrectly added support to optimum-intel:
> 
> * double check how others TTS models are added (Qwen3-TTS, Kokoro). Architecture and the whole structure of optimum-intel should not be broken
> * avoid new test suites added like test_paraformer and others. We should use existing test suites
> * no int4 int8 compression results, GPU results?

Thanks for the review. I’ve pushed a revision that removes the Paraformer plugin and its exporter-function patching, routes export through FunASR registration, and moves Paraformer coverage into the existing OpenVINO test suites. The full-model export tests are opt-in. This PR does not change the OpenVINO exporter entrypoint or CLI command file relative to its base.

For the performance question, short synchronous runs on one input shape measured median latencies of 117.34 ms CPU / 42.30 ms GPU for FP16 and 64.56 ms CPU / 42.79 ms GPU for an existing INT8 IR on ARL
